app/experimenter/static/rapid/components/forms/ExperimentForm.tsx

```diff
@@ -197,6 +197,8 @@ export const SettingsForm: React.FC = () => {
 };
 
 export const BranchesForm: React.FC = () => {
+  const formData = useExperimentState();
+  const variants = formData.variants || [];
```

Comment:
Does the serializer spit out `null` or `[]` if it doesnt' have any variants?

Reply:
good question, I was hoping it should be `[]` – what's the right way to do that?

Reply:
Oh I wasn't even thinking.  Yeah so basically the pattern we've gone with after having to go through this a few times now is for the serializer to spit out an 'empty' state that's valid for the frontend, so we'd make sure that when you pass no instance into the serializer what comes out is valid.  So in this case it should be `variants: []`, which I think will happen automatically with the list serializer, but it's good to assert on.

Which reminds me that we didn't add a testcase for just the ExperimentRapidVariantSerializer by itself, which we should do.

Sorry @tiftran ! Can you pls add a test class for the ExperimentRapidVariantSerializer that tests

- its expected output
- move the ratio validation test in there